pkg/venafi/tpp/connector.go

```diff
@@ -395,6 +509,58 @@ func (c *Connector) RequestCertificate(req *certificate.Request) (requestID stri
 		return "", fmt.Errorf("%s", err)
 	}
 
+	//Handle legacy TPP custom field API
+	if len(req.CustomFields) > 0 {
+		//Get the saved metadata for the current certificate, deep compare the
+		//saved metadata to the requested metadata. If all items match then no further
+		//changes need to be made. If they do not match, they try to update them using
+		//the 19.2 WebSDK calls
+		metadataItems, err := c.requestMetadataItems(requestID)
+		if err != nil {
+			return "", fmt.Errorf("%s", err)
+		}
+
+		//Deep compare the request metadata to the fetched metadata
+		var allItemsFound = true
+	OUTER:
+		for _, cf := range tppCertificateRequest.CustomFields {
```

Comment:
Please rewrite this code. Remove labeled cycles. 
4 nested cycles can replaced with search in prepared map,  helper functions.

Reply:
I am not the best a Go maps. Can you help me understand how that is cleaner and faster?